src/plonk/circuit.rs

```diff
@@ -1147,12 +1147,16 @@ impl<F: Field> ConstraintSystem<F> {
         index
     }
 
-    fn query_any_index(&mut self, column: Column<Any>, at: Rotation) -> usize {
-        match column.column_type() {
-            Any::Advice => self.query_advice_index(Column::<Advice>::try_from(column).unwrap(), at),
-            Any::Fixed => self.query_fixed_index(Column::<Fixed>::try_from(column).unwrap(), at),
+    fn query_any_index<C: Into<Column<Any>> + Copy>(&mut self, column: C, at: Rotation) -> usize {
+        match column.into().column_type() {
+            Any::Advice => {
+                self.query_advice_index(Column::<Advice>::try_from(column.into()).unwrap(), at)
+            }
+            Any::Fixed => {
+                self.query_fixed_index(Column::<Fixed>::try_from(column.into()).unwrap(), at)
+            }
```

Comment:
This diff is much larger than it needs to be. Instead use shadowing, which would turn this into a two-line diff (and also completely remove the need for the `Copy` bound):
```suggestion
        let column = column.into();
        match column.column_type() {
            Any::Advice => {
                self.query_advice_index(Column::<Advice>::try_from(column).unwrap(), at)
            }
            Any::Fixed => {
                self.query_fixed_index(Column::<Fixed>::try_from(column).unwrap(), at)
            }
```